46.md

```diff
@@ -1,98 +1,162 @@
-NIP-46
-======
+# NIP-46 - Nostr Remote Signing
 
-Nostr Connect
--------------
+## Rationale
 
-`draft` `optional`
+Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
 
-This NIP describes a method for 2-way communication between a **remote signer** and a normal Nostr client. The remote signer could be, for example, a hardware device dedicated to signing Nostr events, while the client is a normal Nostr client.
+This NIP describes a method for 2-way communication between a remote signer and a Nostr client. The remote signer could be, for example, a hardware device dedicated to signing Nostr events, while the client is a normal Nostr client.
+
+## Terminology
+
+- **Local keypair**: A local public and private key-pair used to encrypt content and communicate with the remote signer.
+- **Remote pubkey**: The public key that the user wants to sign as. The remote signer has control of the private key that matches this public key.
```

Comment:
"remote keypair" sounds like it could be referring to the signer rather than the user. What about:

```suggestion
- **Client keypair**: A local public and private key-pair generated by the client, used to encrypt content and communicate with the remote signer.
- **Signer pubkey**: The public key belonging to the remote signer. This is either explicitly provided to the client, or discovered by accessing the signer's `/.well-known/nostr.json` file and looking up the `nip05._` entry.
- **User pubkey**: The public key that the user wants to sign as. The remote signer has control of the private key that matches this public key.
```

Reply:
I think this is more confusing. signer pubkey and user pubkey are the same no? no point creating different names.

Reply:
Ok, fair, but your confusion illustrates my point. The signer in this case is the service that custodies the user's key. Maybe "bunker" but that's a pretty idiosyncratic term.

Reply:
I like `signer pubkey` best.

Reply:
Updated the language in the latest.